homeassistant/components/garage_door/myq.py

```diff
@@ -0,0 +1,172 @@
+"""
+Support for Chamberlain garage doors.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/garage_door.myq/
+"""
+
+import logging
+import requests
+
+from homeassistant.components.garage_door import GarageDoorDevice
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD, CONF_NAME
+
+
+REQUIREMENTS = []
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the MyQ garage door."""
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+    name = config.get(CONF_NAME) if \
+        CONF_NAME else "MyQ"  # is this the right way?
+
+    if username is None or password is None:
+        logging.getLogger(__name__).error(
+            "Missing username or password.")
+        return
+
+    myq = MyQAPI(username, password)
+
+    add_devices(MyQGarageDoorDevice(myq, door, name) for door
+                in myq.get_garage_doors())
+
+
+class MyQAPI(object):
+    """Class for interacting with the MyQ iOS App API."""
+
+    # as bananas as it seems, this token is valid for all iOS app instances
+    APP_ID = \
+        "Vj8pQggXLhLy0WHahglCD4N1nAkkXQtGYpq2HrHD7H1nvmbT55KqtN6RSF4ILB%2Fi"
+    LOCALE = "en"
+
+    HOST_URI = "myqexternal.myqdevice.com"
+    LOGIN_ENDPOINT = "Membership/ValidateUserWithCulture"
+    DEVICE_LIST_ENDPOINT = "api/UserDeviceDetails"
+    DEVICE_SET_ENDPOINT = "Device/setDeviceAttribute"
+    DEVICE_STATUS_ENDPOINT = "/Device/getDeviceAttribute"
+    DOOR_STATE = {
+        '1': 'open',
+        '2': 'close',
+        '4': 'opening',
+        '5': 'closing',
+        '8': 'in_transition',
+        '9': 'open'
+    }
+
+    def __init__(self, username, password):
+        """Initialize the API object."""
+        self.username = username
+        self.password = password
+        self.security_token = None
+        self._logged_in = False
+
+    def login(self):
+        """Log in to the service."""
+        params = {
+            'username': self.username,
+            'password': self.password,
+            'appId': self.APP_ID,
+            'culture': 'en'
```

Comment:
Is 'en' generic or could this lead to issues if people in Europe/Asia wants to use it?